src/client.js

```diff
@@ -1936,8 +1936,14 @@ function SugarApi(args) {
                     // this is not our message, ignore it
                     return;
                 }
-                $(window).on('message', null);
+
                 var authData = $.parseJSON(event.originalEvent.data);
+                if (_.isUndefined(authData.external_login) || authData.external_login !== true) {
```

Comment:
Sample message data:

External login:

{"access_token":"3bcaaffb-e096-40c8-a61c-3ff5c764477c","expires_in":3600,"token_type":"bearer","scope":null,"refresh_token":"039707da-7ae2-403f-a8a8-93de4836acf0","refresh_expires_in":1209600,"download_token":"f299ee8d-fc97-465a-ab0c-e6727dfb88c7","external_login":true}

Pendo:
{"apiKey":"1dd345e9-b638-4bd2-7bfb-147a937d4728","frameId":"Grm39NKI2cc5aF9YZWkJyvLmJPgUOBv6","type":"frame:load","guid":"pendo-1"}

Reply:
Base on [Mango PR](https://github.com/sugarcrm/Mango/pull/37197/files). This will properly handle Pendo login, by not treat it as "external_login".

Reply:
If the server always sends the external_login flag, we can reduce the logic here to:
```js
var authData = $.parseJSON(event.originalEvent.data);
if (!authData.external_login) {
    // Not our message, ignore it
    return;
}
```

Reply:
Done